docs/ibc/light-clients/client-state.md

```diff
@@ -1,3 +1,65 @@
 <!--
 order: 2
--->
\ No newline at end of file
+-->
+
+# Implementing the ClientState interface
+
+Learn how to implement the [Client State](https://github.com/cosmos/ibc-go/blob/main/modules/core/exported/client.go#L36) interface.
+
+### ClientType
+
+`ClientType` should return a unique string identifier of the light client.
+
+### GetLatestHeight
+
+`GetLatestHeight` should return the latest block height.
+
+### Validate
+
+`Validate` should validate every client state field and should return an error if any value is invalid. The light client
+implementor is in charge of determining which checks are required. See the [tendermint light client implementation](https://github.com/cosmos/ibc-go/blob/main/modules/light-clients/07-tendermint/client_state.go#L111)
+as a reference.
+
+### Status
+
+`Status` must return the status of the client. Only `Active` clients are allowed to process packets. All
+possible Status types can be found [here](https://github.com/cosmos/ibc-go/blob/main/modules/core/exported/client.go).
+
+### ZeroCustomFields
+
+`ZeroCustomFields` should return a copy of the light client with all client customizable fields with their zero value. It should not mutate the fields of the light client.
+This method is used to [verify upgrades](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/types/proposal.go#L120) and when [scheduling upgrades](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/keeper/proposal.go#L82).
+
+### GetTimestampAtHeight
+
+`GetTimestampAtHeight` must return the timestamp for the consensus state associated with the provided height.
+
+### Initialize
+
+Clients must validate the initial consensus state, and may store any client-specific metadata necessary
+for correct light client operations in the `Initialize` function.
```

Comment:
I think this second line could be removed?

Reply:
to clarify do you mean the line you  commented on or the following line which links to the code where the method gets called?

Reply:
The line I commented on